From ba72de086bac233a8e5d3ed4e47fd3df6f6cd91a Mon Sep 17 00:00:00 2001
From: Song Zhao <song.zhao@amlogic.com>
Date: Wed, 19 Feb 2020 21:43:28 -0800
Subject: [PATCH] v4l2: fix kernel panic after EOS [1/1]

PD#SWPL-20924

Problem:
After EOS, if there are still frames in dpb, vdec_v4l2_buffer can not be
fetched from vframe. Because rest flow already clear all buffer_specs[].

Solution:
Check for valid v4l_mem_handle

Verify:
U212

Change-Id: I6e22f538bbe01b258acf2b2f6cb5411fdc89e1c4
Signed-off-by: Song Zhao <song.zhao@amlogic.com>
---
 drivers/amvdec_ports/decoder/vdec_h264_if.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/amvdec_ports/decoder/vdec_h264_if.c b/drivers/amvdec_ports/decoder/vdec_h264_if.c
index 8f2f9606..4734701c 100644
--- a/drivers/amvdec_ports/decoder/vdec_h264_if.c
+++ b/drivers/amvdec_ports/decoder/vdec_h264_if.c
@@ -730,8 +730,10 @@ static void vdec_h264_get_vf(struct vdec_h264_inst *inst, struct vdec_v4l2_buffe
 	atomic_set(&vf->use_cnt, 1);
 
 	fb = (struct vdec_v4l2_buffer *)vf->v4l_mem_handle;
-	fb->vf_handle = (unsigned long)vf;
-	fb->status = FB_ST_DISPLAY;
+	if (fb) {
+		fb->vf_handle = (unsigned long)vf;
+		fb->status = FB_ST_DISPLAY;
+	}
 
 	*out = fb;
 
-- 
2.24.1

